commands/scanpullrequest.go

```diff
@@ -89,60 +93,77 @@ func scanPullRequest(repo *utils.Repository, client vcsclient.VcsClient, pullReq
+
+	// Download target branch (if needed)
+	targetBranchWd := ""
+	cleanupTarget := func() error { return nil }
+	if repoConfig.IncludeAllVulnerabilities {
+		log.Info("Frogbot is configured to show all vulnerabilities")
+		targetBranchInfo := pullRequestDetails.Target
+		targetBranchWd, cleanupTarget, err = utils.DownloadRepoToTempDir(client, targetBranchInfo.Owner, targetBranchInfo.Repository, targetBranchInfo.Name)
+		if err != nil {
+			return
+		}
+	}
+	defer func() {
+		err = errors.Join(err, cleanupSource(), cleanupTarget())
+	}()
+
+	scanDetails := utils.NewScanDetails(client, &repoConfig.Server, &repoConfig.Git).
+		SetXrayGraphScanParams(repoConfig.Watches, repoConfig.JFrogProjectKey).
+		SetMinSeverity(repoConfig.MinSeverity).
+		SetFixableOnly(repoConfig.FixableOnly)
+
 	for i := range repoConfig.Projects {
-		// Set source branch scan details
-		scanDetails := utils.NewScanDetails(client, &repoConfig.Server, &repoConfig.Git).
-			SetProject(&repoConfig.Projects[i]).
-			SetXrayGraphScanParams(repoConfig.Watches, repoConfig.JFrogProjectKey).
-			SetMinSeverity(repoConfig.MinSeverity).
-			SetFixableOnly(repoConfig.FixableOnly).
-			SetBranch(sourceBranch).
-			SetRepoOwner(pullRequestDetails.Source.Owner)
+		scanDetails.SetProject(&repoConfig.Projects[i])
+
 		// Audit source branch
-		sourceResults, err := downloadAndAuditBranch(scanDetails)
+		var sourceResults *audit.Results
+		sourceResults, err = runInstallAndAudit(scanDetails, sourceBranchWd)
 		if err != nil {
-			return nil, nil, err
+			return
 		}
 
 		// Set JAS output flags
-		contextualAnalysisResultsExists := len(sourceResults.ExtendedScanResults.ApplicabilityScanResults) > 0
-		entitledForJas := sourceResults.ExtendedScanResults.EntitledForJas
-		repoConfig.OutputWriter.SetJasOutputFlags(entitledForJas, contextualAnalysisResultsExists)
+		extendedScanResults := sourceResults.ExtendedScanResults
+		repoConfig.OutputWriter.SetJasOutputFlags(extendedScanResults.EntitledForJas, len(extendedScanResults.ApplicabilityScanResults) > 0)
 
 		// Get all issues that were found in the source branch
 		if repoConfig.IncludeAllVulnerabilities {
-			log.Info("Frogbot is configured to show all vulnerabilities")
-			allIssuesRows, err := getScanVulnerabilitiesRows(sourceResults)
+			var allIssuesRows []formats.VulnerabilityOrViolationRow
+			allIssuesRows, err = getScanVulnerabilitiesRows(sourceResults)
 			if err != nil {
-				return nil, nil, err
+				return
 			}
 			vulnerabilitiesRows = append(vulnerabilitiesRows, allIssuesRows...)
 			iacRows = append(iacRows, xrayutils.PrepareIacs(sourceResults.ExtendedScanResults.IacScanResults)...)
 			continue
 		}
 
 		// Set target branch scan details
-		scanDetails.SetFailOnInstallationErrors(*repoConfig.FailOnSecurityIssues).
-			SetBranch(targetBranch).
-			SetRepoOwner(pullRequestDetails.Target.Owner)
-		targetResults, err := downloadAndAuditBranch(scanDetails)
+		scanDetails.SetFailOnInstallationErrors(*repoConfig.FailOnSecurityIssues)
```

Comment:
It's not related to your changes, but I think we need to set this flag earlier, because in case of "IncludeAllVulnerabilities" it doesn't apply.